ReferenceApp/src/main/java/com/google/ads/interactivemedia/v3/samples/videoplayerapp/truex/TruexAdManager.java

```diff
@@ -0,0 +1,209 @@
+package com.google.ads.interactivemedia.v3.samples.videoplayerapp.truex;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.ViewGroup;
+
+import com.truex.adrenderer.IEventEmitter.IEventHandler;
+import com.truex.adrenderer.TruexAdEvent;
+import com.truex.adrenderer.TruexAdOptions;
+import com.truex.adrenderer.TruexAdRenderer;
+
+import java.util.Map;
+
+/**
+ * This class holds a reference to the true[X] ad renderer and handles all of the event handling
+ * for the example integration application. This class interacts with the video player by resuming
+ * the content when the engagement is complete.
+ */
+public class TruexAdManager {
+    private static final String CLASSTAG = TruexAdManager.class.getSimpleName();
+
+    private PlaybackHandler playbackHandler;
+    private boolean didReceiveCredit;
+    private TruexAdRenderer truexAdRenderer;
+
+    public TruexAdManager(Context context, PlaybackHandler playbackHandler) {
+        this.playbackHandler = playbackHandler;
+
+        didReceiveCredit = false;
+
+        // Set-up the true[X] ad renderer
+        truexAdRenderer = new TruexAdRenderer(context);
+
+        // Set-up the event listeners
+        truexAdRenderer.addEventListener(TruexAdEvent.AD_FETCH_COMPLETED, this.adFetchCompleted);
```

Comment:
is it possible to do this sample without the TruexAdManager
i.e. try and use the TruexAdRenderer directly instead

the main work seems to be the event handling

what if we add a whipcord listener and do a switch statement like in the other ref apps?

that way we log the events by default, put a line or two comment for each event type on each case explaining what it is

the code then collapses somewhat